### PR TITLE
Iter13

### PR DESCRIPTION
**Pull request by ajugalushkin**

### CODE REVIEW

File: internal/errors/errors.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package errors
+
+import (
+	"fmt"
+
+	"github.com/ajugalushkin/url-shortener-version2/internal/dto"
+)
+
+type DuplicateURLError struct {
```
Comment (arttet) [inline internal/errors/errors.go]: SUGGESTION: if I were you, I would wrap errors.

File: migrations/20240420170940_init_db.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE IF NOT EXISTS shorten_urls (
+    short_url VARCHAR(20) NOT NULL PRIMARY KEY,
+    correlation_id VARCHAR(250) NOT NULL DEFAULT '',
+    original_url VARCHAR(250) NOT NULL DEFAULT ''
```
Comment (arttet) [inline migrations/20240420170940_init_db.sql]:
```suggestion
    original_url VARCHAR(2048) NOT NULL DEFAULT ''
```

Comment (arttet) [inline migrations/20240420170940_init_db.sql]:
```suggestion
    original_url VARCHAR(250) NOT NULL
```

File: internal/storage/repository/repository.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	log.Info("repository.Get OK", zap.String("Original URL", shortening.OriginalURL))
+
+	return &shortening, nil
+}
+
+func (r *Repo) GetByURL(ctx context.Context, originURL string) (*dto.Shortening, error) {
+	var shorteningList []dto.Shortening
+
+	err := database.WithTx(ctx, r.db, func(ctx context.Context, tx *sqlx.Tx) error {
+		sb := squirrel.Select("short_url", "correlation_id", "original_url").
```
Comment (arttet) [inline internal/storage/repository/repository.go:100]: CLARIFICATION: please build an execution cost plan of this query

**Review by arttet — CHANGES_REQUESTED** [review]
Please fix the CodeCov action

File: internal/storage/repository/repository.go (hunk 3 above)
Comment (ajugalushkin) [reply internal/storage/repository/repository.go:100]: image
If a specific URL is specified, I don't see a problem with the request

**Review by arttet — APPROVED** [review]
Please fix the CodeCov action

**Pull request merged**